Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm

```diff
@@ -512,7 +512,7 @@ - (CFURLRef)_cfurl;
     if (applyTrackingPrevention == ApplyTrackingPrevention::Yes && shouldBlockCookies(firstParty, url, frameID, pageID, shouldRelaxThirdPartyCookieBlocking))
         return;
 
-    NSURL *cookieURL = url;
+    RetainPtr<NSURL> cookieURL = static_cast<NSURL *>(url);
```

Comment:
Here the same “write a cast” thing comes up but we don’t have a function like `protectedNSString`. Maybe we should add a `createNSURL` function that returns a `RetainPtr`; nice to avoid the autorelease too. Also makes me realize we could just rename `protectedNSString` to `createNSString`.